keps/sig-api-machinery/0006-apply.md

```diff
@@ -194,8 +195,41 @@ atomic. That can be specified with `// +mapType=atomic` or `//
 +structType=atomic` respectively. They map to the same openapi extension:
 `"x-kubernetes-map-type": "atomic"`.
 
-#### Kubectl
-##### Server-side Apply
+#### Clients
+
+##### Apply Status Header
+
+Client side apply in kubectl determines if the applied object would result in any changes on the apiserver and provides the user with this information.
+
+To provided similar information with server-side apply, a dedicated header is set by the apiserver
+if a **successful apply request** caused changes to the existing resource.
+
+```txt
+X-Kubernetes-Apply-Success-Type: "changes-made" || "changes-unneeded"
+```
+
+The value may be,
+
+- `changes-unneeded` in cases where the request did not cause a change to the resource's managedFields.
+This means changes to fields that don't get owned or get reset, will be ignored when setting the header.
```

Comment:
will it be a problem to return unneeded when the user changed fields either stripped from managedFields or wiped by the not implemented yet status wiping feature?

Reply:
That should be fine, I think.

Reply:
See my comment above.

Reply:
I would just omit this line completely. "changes" that don't actually affect the object are not changes. This may be a difference from what the client side tool does, but I'd argue that the client side tool is reporting a different (and less useful) thing.